File: core/src/ru/myitschool/dcrawler/effects/CloudOfKnivesEffect.java
package ru.myitschool.dcrawler.effects;

import com.badlogic.gdx.graphics.Texture;
import ru.myitschool.dcrawler.entities.Entity;
import ru.myitschool.dcrawler.math.MathAction;

/**
 * Created by Voyager on 29.11.2017.
 */
public class CloudOfKnivesEffect extends CellEffect {

    private String name = "Cloud of knives";
    private String description = "";
    private Texture icon = new Texture("cloud_knives.png");
    private boolean positive = false;
    private MathAction damage;

    public CloudOfKnivesEffect(FloorEffect floorEffect, MathAction damage) {
        super(floorEffect);
        this.damage = damage;
    }

    @Override
    protected void onStepTo(Entity entity) {
        super.onStepTo(entity);
        entity.addHp(-damage.act());
    }

    @Override
    public Texture getIcon() {
        return icon;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getDescription() {
        return description;
    }

    @Override
    public boolean isSkillUse() {
        return false;
    }

    @Override
    public boolean isPositive() {
        return false;
    }

    @Override
    public boolean isStackable() {
        return false;
    }

    @Override
    public int getStackSize() {
        return 0;
    }

    @Override
    public boolean isExpiring() {
        return false;
    }

    @Override
    public int getExpireTurns() {
        return 0;
    }

    @Override
    public String getType() {
        return "main.dcrawler.effect.cloud_of_knives";
    }

    @Override
    public void startTurn() {
        super.startTurn();
        int dmg = -damage.act();
        System.out.println("DAMAGE!!! - " + dmg);
        getEntity().addHp(dmg);
    }
}

File: core/src/com/wolg_vlad/dcrawler/effects/CloudOfKnivesEffect.kt
package com.wolg_vlad.dcrawler.effects

import com.badlogic.gdx.graphics.Texture
import com.wolg_vlad.dcrawler.entities.Entity
import com.wolg_vlad.dcrawler.math.MathAction

/**
 * Created by Voyager on 29.11.2017.
 */
class CloudOfKnivesEffect(floorEffect: FloorEffect?, private val damage: MathAction?) : CellEffect(floorEffect) {
    override val name = "Cloud of knives"
    override val description = ""
    override val icon = Texture("cloud_knives.png")
    override val positive: Boolean = false
    override val skillUse: Boolean = false
    override val stackable: Boolean = false
    override val stackSize: Int = 0
    override val expiring: Boolean = false
    override val id: String = "main.dcrawler.effect.cloud_of_knives"
    override val expireTurns: Int = 0

    override fun onStepTo(entity: Entity) {
        super.onStepTo(entity)
        entity.addHp(-damage!!.act())
    }

    override fun startTurn() {
        super.startTurn()
        val dmg = -damage!!.act()
        entity!!.addHp(dmg)
    }
}